integrations/airflow/openlineage/client.py

```diff
@@ -0,0 +1,58 @@
+import os
+from urllib.parse import urljoin
+
+import attr
+from requests import Session
+from requests.adapters import HTTPAdapter
+
+from openlineage import constants
+from openlineage.run import RunEvent, Serde
+
+
+@attr.s
+class OpenLineageClientOptions:
+    timeout: float = attr.ib(default=5.0)
+    verify: bool = attr.ib(default=True)
+    api_key: str = attr.ib(default=None)
+    adapter: HTTPAdapter = attr.ib(default=None)
+
+
+class OpenLineageClient:
+    def __init__(
+            self,
+            url: str,
+            options: OpenLineageClientOptions = OpenLineageClientOptions(),
+            session: Session = None
+    ):
+        self.url = url  # TODO: url validation?
```

Comment:
Would we just validate that the set value for `url` is a valid URL?

Reply:
Done using `urllib.urlparse`

Reply:
It's not a check I'd normally make, but as long as we silently handle, then log the exception I think this is ok to add.